extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java

```diff
@@ -851,19 +858,28 @@ public void verifyResults()
       Preconditions.checkArgument(expectedDataSource != null, "dataSource cannot be null");
       Preconditions.checkArgument(expectedRowSignature != null, "expectedRowSignature cannot be null");
       Preconditions.checkArgument(
-          expectedResultRows != null || expectedMSQFault != null,
-          "atleast one of expectedResultRows or expectedMSQFault should be set to non null"
+          expectedResultRows != null || expectedMSQFault != null || expectedMSQFaultClass != null,
+          "atleast one of expectedResultRows, expectedMSQFault or expectedMSQFaultClass should be set to non null"
       );
       Preconditions.checkArgument(expectedShardSpec != null, "shardSpecClass cannot be null");
       readyToRun();
       try {
         String controllerId = runMultiStageQuery(sql, queryContext);
-        if (expectedMSQFault != null) {
+        if (expectedMSQFault != null || expectedMSQFaultClass != null) {
           MSQErrorReport msqErrorReport = getErrorReportOrThrow(controllerId);
-          Assert.assertEquals(
-              expectedMSQFault.getCodeWithMessage(),
-              msqErrorReport.getFault().getCodeWithMessage()
-          );
+          if (expectedMSQFault != null) {
+            Assert.assertEquals(
+                expectedMSQFault.getCodeWithMessage(),
+                msqErrorReport.getFault().getCodeWithMessage()
+            );
+          }
+          if (expectedMSQFaultClass != null) {
```

Comment:
Shouldn't we always assert the MSQ fault? 
Is there any specific reason to assert on the MSQ class?

Reply:
With the unparseable exceptions, the exception message contained nonstandard characters (from the erroneous line) which I didn't want to use in the code file. Moreover, this would prevent tests from failing if the input file is changed.

Reply:
We change the get message call at line 863 to be a pattern matcher. That way you need not code out the whole string.

Reply:
Or change the source test file.

Reply:
This got me thinking, if we have a malformed line, we can have a parseException with x mb's  of input. 
Multiple such lines would blow up our report.
Should we chomp the input to lets day 300 bytes or something in ParseException#75 
cc @jon-wei

Reply:
Stack traces of innocuous warnings and errors can also cause a similar issue. Should this be taken up as a separate PR? Also, 300 bytes seem less to me, something like 4KB might work better, wdyt? (I think we should be fine with an even larger limit as the number of warnings sent to the controller are limited).

Reply:
Yup We can do this outside this PR.